components/ImageModal/ImageModal.tsx

```diff
@@ -1,17 +1,58 @@
-import { FC, MouseEvent } from 'react';
+import { FC, MouseEvent, useEffect } from 'react';
 
 import styles from './ImageModal.module.scss';
 
-export const ImageModal: FC = () => {
+export const IMAGE_MODAL = {
+  CONTAINER_ID: 'imgModal',
+  CONTENT_ID: 'modalContent',
+};
+
+interface IImageModalProps {
+  images: HTMLElement[];
+}
+
+export const ImageModal: FC<IImageModalProps> = ({ images = [] }) => {
   const closeModal = (e: MouseEvent<HTMLDivElement>): void => {
     const target = e.currentTarget as HTMLDivElement;
     target.style.display = 'none';
   };
 
+  const zoomOnClick = (e: Event): void => {
+    const target = e.target as HTMLElement;
+    const clonedImg = target.cloneNode(true);
+    const imgModal = document.querySelector('#' + IMAGE_MODAL.CONTAINER_ID) as HTMLElement;
+    const imgModalContent = imgModal.querySelector('#' + IMAGE_MODAL.CONTENT_ID) as HTMLElement;
+    imgModalContent.innerHTML = '';
+    imgModalContent.appendChild(clonedImg);
+    imgModal.style.display = 'block';
+  };
+
+  useEffect(() => {
```

Comment:
You have to use one useEffect and implement dependency array for useEffect

Reply:
Fixed